pysys-examples/internal/testcases/PySys_internal_024/run.py

```diff
@@ -5,20 +5,24 @@
 class PySysTest(BaseTest):
 
 	def execute(self):
+		self.log.info('Using python from     %s', sys.executable)
+		self.log.info('With python libs from %s', os.__file__)
+
 		script = "%s/environment.py" % self.input
 		env = {}
 		env["PYSYS-USER"] = "Simon Batty"
 		env["PYSYS-TEST"] = "Test variable"
 		env["EMPTY-ENV"] = ""
 		env["INT-ENV"] = "1"
 
-		env['PYTHONHOME'] = sys.prefix
+		# this mirrors the logic we use in createEnvirons for sys.executable
 		if PLATFORM=='win32':
 			# on win32, minimal environment must have SYSTEMROOT set
 			env["SYSTEMROOT"] = os.environ["SYSTEMROOT"]
 		else:
 			# On UNIX we may need the python shared libraries on the LD_LIBRARY_PATH
 			env[LIBRARY_PATH_ENV_VAR] = os.environ.get(LIBRARY_PATH_ENV_VAR,'')
```

Comment:
This probably doesn't cause any actual problem in practice, but this will (I think) add a new envar with an empty value even if it didn't exist in the parent environment.  Do any of the supported OSes treat those cases differently?

Reply:
Yes, good idea - I've changed to a cautious approach of stripping extra os.pathsep characters, and only setting it if not an empty string